jenkins-slaves/templates/jenkins-slave-generic-template.yml

```diff
@@ -0,0 +1,94 @@
+---
+kind: Template
+apiVersion: v1
+metadata:
+  name: "${NAME}"
+  annotations:
+    openshift.io/display-name: Generic Build Pod
+    description: "${NAME} build pod template pre-configured to use a jenkins slave in the
+      same project/namespace"
+objects:
+- apiVersion: v1
+  kind: ImageStream
+  metadata:
+    labels:
+      build: "${NAME}"
+    name: jenkins-slave-base-rhel7
+  spec:
+    tags:
+    - from:
+        kind: DockerImage
+        name: openshift3/jenkins-slave-base-rhel7:${SLAVE_IMAGE_TAG}
+      generation: 2
+      importPolicy: {}
+      name: latest
+      referencePolicy:
+        type: Source
```

Comment:
@pabrahamsson This is the ImageStream for the slave that is being built. It also has the key label `role: jenkins-slave` so that it will automatically be added as a slave in Jenkins